packages/fixie/src/agent.ts

```diff
@@ -684,8 +655,11 @@ export class FixieAgent {
 
     // The tunnel may yield different URLs over time. We need to create a new
     // agent revision each time.
-    for await (const currentUrl of deploymentUrlsIter) {
-      await waitForAgentToBeReady();
+    for await (const [currentUrl, runtimeParametersSchema] of this.zipAsyncIterables(
+      deploymentUrlsIter,
+      schemaGenerator
+    )) {
```

Comment:
I feel like we should just be using event emitters but I don't think it's worth trying to change now.

Reply:
Probably -- I briefly started down that path but (probably incorrectly) convinced myself it wasn't going to be significantly better.